### PR TITLE
fix that contents couldn't be updated when they had progress data

### PR DESCRIPTION
**Pull request by PaulBredl**
## Description of changes
Fixed the bug. Solution is weird but ok.

  
## How has this been tested?
Please describe the test strategy you followed.

- [x] automated unit test
- [x] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/content_service/persistence/dao/ContentMetadataEmbeddable.java [hunk 1]
```diff
@@ -29,12 +29,8 @@ public class ContentMetadataEmbeddable {
...
 
     @EqualsAndHashCode.Exclude
-    @ManyToMany(fetch = FetchType.LAZY)
-    @JoinTable(
-            name = "content_tags",
-            joinColumns = @JoinColumn(name = "content_id"),
-            inverseJoinColumns = @JoinColumn(name = "tag_id")
-    )
+    @ManyToMany(fetch = FetchType.LAZY, cascade =
+            {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/content_service/persistence/dao/ContentMetadataEmbeddable.java:33]: I also had some problems with the tags. This seems to resolve it

File: src/main/java/de/unistuttgart/iste/gits/content_service/persistence/dao/UserProgressDataEntity.java [hunk 2]
```diff
@@ -17,15 +17,16 @@
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-@IdClass(UserProgressDataPrimaryKey.class)
 public class UserProgressDataEntity {
 
     @Id
-    @Column(name = "user_id", nullable = false)
+    @GeneratedValue
+    private UUID id;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/content_service/persistence/dao/UserProgressDataEntity.java:24]: I tried many other things, but this is the only thing that worked: having a new id field for the entities, even though it is completely unecessary

**Review by myluki2000 — APPROVED** [review]
yay you did it
